File: lectures/Python-module/04_Boost_Python/_7_wrapping_templates.py
from _7_wrapping_templates import Foo_int,Foo_float,Foo_double

f = Foo_int()
print(f.value)
f = Foo_float()
print(f.value, "  <- float has ~7 mantissa significant digits ") 
f = Foo_double()
print(f.value, "  <- double has ~16 mantissa significant digits ") 

File: lectures/Python-module/04_Boost_Python/_7_wrapping_templates.cc
#include <iostream>

template <typename T> 
struct Foo {
	Foo(): value{(T) 0.01234567890123456789} {}
	T value;
};

#include <boost/python.hpp>
using namespace boost::python;

BOOST_PYTHON_MODULE(_7_wrapping_templates) {
	class_<Foo <int> >("Foo_int")
		.def_readonly("value", &Foo <int>::value); 	
	class_<Foo <float> >("Foo_float")
		.def_readonly("value", &Foo <float>::value); 	
	class_<Foo <double> >("Foo_double")
		.def_readonly("value", &Foo <double>::value); 	
} 


